lib/AST/Identifier.cpp

```diff
@@ -79,9 +96,29 @@ int Identifier::compare(Identifier other) const {
   return str().compare(other.str());
 }
 
+DeclName DeclName::createSubscript(ASTContext &C,
+                                   ArrayRef<Identifier> argumentNames) {
+  return DeclName(C, Identifier(SubscriptIdentifierData), argumentNames);
+}
+
+DeclName DeclName::createSubscript() {
+  return Identifier(SubscriptIdentifierData);
+}
+
+DeclNameKind DeclName::getKind() const {
```

Comment:
Sorry, `static` as a member doesn't mean the same thing as `static` elsewhere. As long as it's defined out of line there's no problem declaring it ahead of time.